en/docs/guides/user-mgt/add-user-roles.md

```diff
@@ -0,0 +1,284 @@
+
+WSO2 supports role-based authentication model where privileges of a user are based on the role it is attached to. By default, WSO2 Identity Server comes with the following roles:
+
+- Admin - Provides full access to all features and controls. By default, the user "admin" is assigned to both the "Admin" and the "Everyone" roles. The Identity Server does not allow you to configure permissions assigned to the 'Admin' role.
+- Internal/Everyone - Every new user is assigned to this role by default. It does not include any permission.
+- System - This role is not visible for modification in the interface. 
+
+If a user has several roles assigned, their permissions are added together.
+
+WSO2 products has two types of roles. **External Roles** and **Internal
+Roles**. Assume there are two user stores.
+
+<table>
+<thead>
+<tr class="header">
+<th><br />
+</th>
+<th><strong>Store-A</strong></th>
+<th><strong>Store-B</strong></th>
+</tr>
+</thead>
+<tbody>
+<tr class="odd">
+<td>Users</td>
+<td>user_A</td>
+<td>user_B</td>
+</tr>
+<tr class="even">
+<td>Roles</td>
+<td>role_A</td>
+<td>role_B</td>
+</tr>
+</tbody>
+</table>
+
+**External Roles :**  
+These roles are stored in the user stores itself. Only users that belong to a user store can be assigned to
+external roles in that user store.  
+Example : user\_A can be assigned to role\_A while user\_B cannot be assigned to role\_A.
+
+In user stores, users are assigned to **groups**. Within WSO2 IS, one user group is directly mapped to a role and permissions are assigned to the role.
+  
+**Internal Roles :**
+
+These roles are stored in the WSO2 Identity Server database. Users in all user stores can be assigned to
+these roles.
+
+In the above example, both user\_A and user\_B can be assigned to the same internal role.
+
+For internal roles, there are no mapped groups in the user stores. So we
+directly assign users to these roles.
+
+!!! info 
+    **Internal/everyone :** This is a conceptual role that is used to group
+    all the users (across the user stores) together. When a new user is created, the `internal everyone` role is automatically assigned to the user.
+
+    **Application Role :** This role is used for special cases where an internal role is created for a single service provider (SP) application and only users that are assigned to this role can manage the relevant SP application.
+
+---
+
+## Adding a user role
+
+### Adding a user role using the admin portal
+
+{!fragments/xxx!}
+
+### Adding a user role using SCIM
+
+In SCIM, creating a role is the same as creating a **group.** For more information, see
+    [SCIM](insert-scim2-concept).
+
+!!! note
+    To create a group with users, the said users should already exist in the user store.
+    
+
+Create a new group named **engineer** with the user 'Mark' as a member. The attributes you have to include in the cURL command are the `userID`, `username:password`.
```

Comment:
Since this is a generic guide and not a sample, shall we stick to giving instructions in a generic manner? WDYT:

Use the following curl command to create a new user group with a new member. The attributes you have to include in the cURL command are the `userID`, `username:password`. 

The sample request given below adds a group named "engineer" with the user "Mark" as a member.